tools/codegen/api/lazy.py

```diff
@@ -162,36 +188,37 @@ def aten_name(self) -> str:
     def base_name(self) -> str:
         return f"{self.name.name.base}"
 
-    def filtered_types(self, positional: bool = True, keyword: bool = True,
-                       values: bool = True, scalars: bool = True) -> List[NamedCType]:
-        types: List[NamedCType] = []
+    def filtered_args(self, positional: bool = True, keyword: bool = True,
```

Comment:
The filtering logic is pretty complicated - maybe add a comment above this describing what its purpose is and what it filters?

(sorry for giving feedback on parts of the code that aren't changed directly by the PR 😛 )

Reply:
yea, that's fair. I am totally game to add better comments here.  But if you have other ideas for improvement don't hold back.